packages/duckdb-wasm/src/bindings/bindings_base.ts

```diff
@@ -222,13 +222,19 @@ export abstract class DuckDBBindingsBase implements DuckDBBindings {
         return this.mod.ccall('duckdb_web_pending_query_cancel', 'boolean', ['number'], [conn]);
     }
     /** Fetch query results */
-    public fetchQueryResults(conn: number): Uint8Array {
+    public fetchQueryResults(conn: number): Uint8Array | null {
         const [s, d, n] = callSRet(this.mod, 'duckdb_web_query_fetch_results', ['number'], [conn]);
         if (s !== StatusCode.SUCCESS) {
             throw new Error(readString(this.mod, d, n));
         }
+
         const res = copyBuffer(this.mod, d, n);
         dropResponseBuffers(this.mod);
+
+        // Special case indicating to the caller they need to retry
+        if (res.length === 1 && res[0] == 84) {
+            return null;
+        }
```

Comment:
Alternative would be returning an arrow error with custom error code, unsure if that is properly supported, let me check.

Reply:
Yeah that's the other option we considered, and I agree it should work just as well. We felt this way was the most straightforward and explicit compared to abuse a specific Arrow error code.